peer/config.go

```diff
@@ -0,0 +1,228 @@
+
+	// ChannelDB is used to fetch opened channels, closed channels, and the
+	// channel graph.
+	ChannelDB *channeldb.DB
+
+	// ChainArb is used to subscribe to channel events, update contract signals,
+	// and force close channels.
+	ChainArb *contractcourt.ChainArbitrator
+
+	// AuthGossiper is needed so that the Brontide impl can register with the
+	// gossiper and process remote channel announcements.
+	AuthGossiper *discovery.AuthenticatedGossiper
+
+	// ChanStatusMgr is used to set or un-set the disabled bit in channel
+	// updates.
+	ChanStatusMgr *netann.ChanStatusManager
+
+	// ChainIO is used to retrieve the best block.
+	ChainIO lnwallet.BlockChainIO
+
+	// FeeEstimator is used to compute our target ideal fee-per-kw when
+	// initializing the coop close process.
+	FeeEstimator chainfee.Estimator
+
+	// Signer is used when creating *lnwallet.LightningChannel instances.
+	Signer input.Signer
+
+	// SigPool is used when creating *lnwallet.LightningChannel instances.
+	SigPool *lnwallet.SigPool
+
+	// Wallet is used to publish transactions and generate delivery scripts
+	// during the coop close process.
+	Wallet *lnwallet.LightningWallet
+
+	// ChainNotifier is used to receive confirmations of a coop close
+	// transaction.
+	ChainNotifier chainntnfs.ChainNotifier
+
+	// RoutingPolicy is used to set the forwarding policy for links created by
+	// the Brontide.
+	RoutingPolicy htlcswitch.ForwardingPolicy
+
+	// Sphinx is used when setting up ChannelLinks so they can decode sphinx
+	// onion blobs.
+	Sphinx *hop.OnionProcessor
+
+	// WitnessBeacon is used when setting up ChannelLinks so they can add any
+	// preimages that they learn.
+	WitnessBeacon contractcourt.WitnessBeacon
+
+	// Invoices is passed to the ChannelLink on creation and handles all
+	// invoice-related logic.
+	Invoices *invoices.InvoiceRegistry
+
+	// ChannelNotifier is used by the link to notify other sub-systems about
+	// channel-related events and by the Brontide to subscribe to
+	// ActiveLinkEvents.
+	ChannelNotifier *channelnotifier.ChannelNotifier
+
+	// HtlcNotifier is used when creating a ChannelLink.
+	HtlcNotifier *htlcswitch.HtlcNotifier
+
+	// TowerClient is used when creating a ChannelLink.
+	TowerClient wtclient.Client
+
+	// DisconnectPeer is used to disconnect this peer if the cooperative close
+	// process fails.
+	DisconnectPeer func(*btcec.PublicKey) error
+
+	// GenNodeAnnouncement is used to send our node announcement to the remote
+	// on startup.
+	GenNodeAnnouncement func(bool,
+		...netann.NodeAnnModifier) (lnwire.NodeAnnouncement, error)
+
+	// PrunePersistentPeerConnection is used to remove all internal state
+	// related to this peer in the server.
+	PrunePersistentPeerConnection func([33]byte)
+
+	// FetchLastChanUpdate fetches our latest channel update for a target
+	// channel.
+	FetchLastChanUpdate func(lnwire.ShortChannelID) (*lnwire.ChannelUpdate,
+		error)
+
+	// ProcessFundingOpen is used to hand off an OpenChannel message to the
```

Comment:
Non-blocking, but in the future we could break these up into embedded sub-configs to make this less of a "mega struct".

Reply:
Yeah I was thinking a FundingManager interface could be exposed in a later PR?